sentry/sentry.go

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package sentry
+
+import (
+	"os"
+	"time"
+
+	raven "github.com/getsentry/raven-go"
+	"github.com/pkg/errors"
+	"github.com/uber-go/zap"
+	"github.com/uber-go/zap/zwrap"
+)
+
+var (
+	_platform          = "go"
+	_traceContextLines = 3
+	_traceSkipFrames   = 3
+	_zapToRavenMap     = map[zap.Level]raven.Severity{
+		zap.InfoLevel:  raven.INFO,
+		zap.WarnLevel:  raven.WARNING,
+		zap.ErrorLevel: raven.ERROR,
+		zap.PanicLevel: raven.ERROR,
+		zap.FatalLevel: raven.FATAL,
+	}
+)
+
+// Logger automatically sends logs above a certain threshold to Sentry.
+type Logger struct {
```

Comment:
Do we need to export this type, or can `New` just return a `zap.Logger`? If possible, I'd prefer that - you can cast back to a `*logger` in your tests.